FlowPlugins/CommunityFlowPlugins/tools/notifySonarrRadarrLidarr/1.0.1/index.js

```diff
@@ -0,0 +1,142 @@
+exports.details = details;
+var plugin = function (args) { return __awaiter(void 0, void 0, void 0, function () {
+    var lib, _a, arr, arr_api_key, arr_host, fileName, arrHost, headers, requestConfig, res, movieId, requestConfig2, requestConfig, res, seriesId, requestConfig2, artistId, requestConfig;
+    return __generator(this, function (_d) {
+        switch (_d.label) {
+            case 0:
+                lib = require('../../../../../methods/lib')();
+                args.inputs = lib.loadDefaultValues(args.inputs, details);
+                _a = args.inputs, arr = _a.arr, arr_api_key = _a.arr_api_key, arr_host = _a.arr_host;
+                fileName = args.originalLibraryFile.meta.FileName || '';
+                arrHost = arr_host.endsWith('/') ? arr_host.slice(0, -1) : arr_host;
+                headers = { 'Content-Type': 'application/json', 'X-Api-Key': arr_api_key, Accept: 'application/json' };
+                args.jobLog('Going to force scan');
+                if (!(arr === 'radarr')) return [3 /*break*/, 3];
+                args.jobLog('Refreshing Radarr...');
+                requestConfig = { method: 'get', url: "".concat(arrHost, "/api/v3/parse?title=").concat(encodeURIComponent(fileName)), headers: headers };
+                return [4 /*yield*/, args.deps.axios(requestConfig)];
+            case 1:
+                res = _d.sent();
+                movieId = res.data.movie.movieFile.movieId;
+                requestConfig2 = { method: 'post', url: "".concat(arrHost, "/api/v3/command"), headers: headers, data: JSON.stringify({ name: 'RefreshMovie', movieIds: [movieId] }) };
+                return [4 /*yield*/, args.deps.axios(requestConfig2)];
+            case 2:
+                _d.sent();
+                args.jobLog("\u2714 Refreshed movie ".concat(movieId, " in Radarr."));
+                return [3 /*break*/, 10];
+            case 3:
+                if (!(arr === 'sonarr')) return [3 /*break*/, 6];
+                args.jobLog('Refreshing Sonarr...');
+                requestConfig = { method: 'get', url: "".concat(arrHost, "/api/v3/parse?title=").concat(encodeURIComponent(fileName)), headers: headers };
+                return [4 /*yield*/, args.deps.axios(requestConfig)];
+            case 4:
+                res = _d.sent();
+                seriesId = res.data.series.id;
+                requestConfig2 = { method: 'post', url: "".concat(arrHost, "/api/v3/command"), headers: headers, data: JSON.stringify({ name: 'RefreshSeries', seriesId: seriesId }) };
+                return [4 /*yield*/, args.deps.axios(requestConfig2)];
+            case 5:
+                _d.sent();
+                args.jobLog("\u2714 Refreshed series ".concat(seriesId, " in Sonarr."));
+                return [3 /*break*/, 10];
+            case 6:
+                if (!(arr === 'lidarr')) return [3 /*break*/, 9];
+                args.jobLog('Refreshing Lidarr...');
+                artistId = 'artistId'; // Placeholder for artist ID
```

Comment:
Does this even work? Radarr & Sonarr need the IDs, a placeholder for Lidarr seems a bit odd.